### PR TITLE
middleware, route and controllers

### PR DESCRIPTION
**Pull request by floor114**

### CODE REVIEW

**Comment by floor114** [comment]
Ще треба доробити моделі і написати тести

File: 05/app/controllers/heroes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    render :show
+  end
+
+  def new
+    render 'heroes/new'
+  end
+
+  def create
+    hero = Hero.new(
+      name: params['hero']['name'],
```
Comment (kushniryb) [inline 05/app/controllers/heroes_controller.rb]: Seems kinda ugly
Why not something likes this
```ruby

def hero_params
  params['hero']
end

def create
  hero = Hero.new(hero_params)
  ....
end
```

File: 05/app/middleware/authenticator.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Authenticator
+  PASSWORD = 'ILoveRuby'.freeze
+
+  def initialize(app)
+    @app = app
+    self
+  end
+
+  def call(env)
+    if (env['HTTP_PASSWORD'] || '').strip == PASSWORD
```
Comment (kushniryb) [inline 05/app/middleware/authenticator.rb]: Extract to separate method according to SRP

**Comment by kushniryb** [comment]
LGTM overall :+1:

**Pull request merged**